activerecord/lib/active_record/query_logs.rb

```diff
@@ -130,7 +131,7 @@ def tag_content
             else
               handler
             end
-            "#{key}:#{val}" unless val.nil?
+            "#{key}='#{val.gsub("'", "\\\\'")}'" unless val.nil?
```

Comment:
This change introduces a requirement that `val` must be a string, which is documented above in the comments. I opted for having this requirement because I think the logic is more readable/maintainable, but I'm open to feedback of course. An alternative is to call `.to_s` on `val` here, or perhaps adding a check to only call `.gsub` when val is a string.